tests/flit_cli/flit_init/tst_litmustests.py

```diff
@@ -0,0 +1,122 @@
+#
+# - Redistributions in binary form must reproduce the above
+#   copyright notice, this list of conditions and the disclaimer
+#   (as noted below) in the documentation and/or other materials
+#   provided with the distribution.
+#
+# - Neither the name of the LLNS/LLNL nor the names of its
+#   contributors may be used to endorse or promote products derived
+#   from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+# CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL
+# SECURITY, LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+# TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+# THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Additional BSD Notice
+#
+# 1. This notice is required to be provided under our contract
+#    with the U.S. Department of Energy (DOE). This work was
+#    produced at Lawrence Livermore National Laboratory under
+#    Contract No. DE-AC52-07NA27344 with the DOE.
+#
+# 2. Neither the United States Government nor Lawrence Livermore
+#    National Security, LLC nor any of their employees, makes any
+#    warranty, express or implied, or assumes any liability or
+#    responsibility for the accuracy, completeness, or usefulness of
+#    any information, apparatus, product, or process disclosed, or
+#    represents that its use would not infringe privately-owned
+#    rights.
+#
+# 3. Also, reference herein to any specific commercial products,
+#    process, or services by trade name, trademark, manufacturer or
+#    otherwise does not necessarily constitute or imply its
+#    endorsement, recommendation, or favoring by the United States
+#    Government or Lawrence Livermore National Security, LLC. The
+#    views and opinions of authors expressed herein do not
+#    necessarily state or reflect those of the United States
+#    Government or Lawrence Livermore National Security, LLC, and
+#    shall not be used for advertising or product endorsement
+#    purposes.
+#
+# -- LICENSE END --
+
+'''
+Test FLiT init --litmus-test argument
+
+These tests use doctest
+
+>>> import os
+>>> import subprocess as subp
+>>> import tempfile
+
+Call init --litmus-tests into a temporary directory
+  We suppress console output here
+>>> with th.tempdir() as temp_dir:
+...     fname = os.path.join(temp_dir, 'tests/')
+...     null = subp.check_call(['flit', 'init', '-C', temp_dir, '--litmus-tests'])
```

Comment:
This will not work on systems without flit installed.  It is better to call using the test harness, e.g., `th.flit.main(['init', '-C', temp_dir, '--litmus-tests'])`

Reply:
You may also want to capture the standard output into a variable.  See lines 99-108 in `tests/flit_cli/flit_bisect/tst_bisect_autosqlite_clang.py`.  I've copied those lines here for reference:

```python
>>> class BisectTestError(RuntimeError): pass

>>> with th.tempdir() as temp_dir:
...     with StringIO() as ostream:
...         retval = th.flit.main(['init', '-C', temp_dir], outstream=ostream)
...         if retval != 0:
...             raise BisectTestError(
...                 'Could not initialize (retval={0}):\\n'.format(retval) +
...                 ostream.getvalue())
...         init_out = ostream.getvalue().splitlines()
```

The `retval` variable is the return value from calling `main()`.  It is supposed to return 0 if it finished successfully, which is what is being checked here.  If you want to throw a different exception, or skip the error checking that's fine - it's your choice.

Reply:
One last comment:

You store the output into a variable called `null`.  The "pythonic" way to ignore an output value is to store it into the `_` variable.  e.g.,

```python
_ = subp.check_call([...])
```